methods/basic.go

```diff
@@ -15,7 +20,62 @@ func BasicRequestWithBody(c *cli.Context, method string) (string, error) {
 		return "", err
 	}
 
-	var jsonStr = []byte(c.String("body"))
+	var jsonStr []byte
+
+	if c.Bool("editor") {
+		var path string
+		var editor string
+
+		// check OS
+		currentOs := runtime.GOOS
+
+		// based on OS find assign default editor
+		if currentOs == "linux" || currentOs == "darwin" {
+			editor = "nano"
+		} else if currentOs == "windows" {
+			editor = "notepad"
+		}
+
+		path, err = exec.LookPath(editor)
+		if err != nil {
+			return "", fmt.Errorf("Unable to locate %s at %s", editor, path)
+		}
+
+		// create temp file for writing request body
+		tempFile, err := ioutil.TempFile("", "HOPP-CLI-REQUEST*.txt")
+		if err != nil {
+			return "", fmt.Errorf("Unable to create temp file: %s\n%s", tempFile.Name(), err)
+		}
+
```

Comment:
`defer os.Remove(tempFile.Name())` should move to just after the error check. Otherwise, if an error condition is reached before the deferred deletion, you'll pollute the filesystem with empty temp files.

The convention in Go is to place cleanup `defer` statements immediately after the error check.